### PR TITLE
secret pkg small refactoring + WithStringData method

### PR DESCRIPTION
**Pull request by elenagerman**

### CODE REVIEW

**Comment by elenagerman** [comment]
@trewest @kononovn @achuzhoy @cdvultur PTAL

File: pkg/configmap/configmap_test.go [hunk 1]
```diff
@@ -196,20 +197,34 @@ func TestDelete(t *testing.T) {
...
-
-	assert.Equal(t, "", testBuilder.errorMsg)
+func TestUpdate(t *testing.T) {
+	testCases := []struct {
+		testCM        *Builder
+		testData      map[string]string
+		expectedError error
+	}{
+		{
+			testCM:        buildTestBuilderWithFakeObjects([]runtime.Object{}),
```
Comment (trewest) [inline pkg/configmap/configmap_test.go:207]: Please add a test where a configmap resource is added to runtime objects

Comment (elenagerman) [reply pkg/configmap/configmap_test.go:207]: @trewest this is secret, not configmap.
do you ask me to add runtime test to the "With..." method?

Comment (trewest) [reply pkg/configmap/configmap_test.go:207]: I think it is an issue with either the title or contents then? The title says secret pkg but it looks like you updated configmap files

File: pkg/configmap/configmap.go [hunk 2]
```diff
@@ -157,6 +157,32 @@ func (builder *Builder) Exists() bool {
 	return err == nil || !k8serrors.IsNotFound(err)
 }
 
+// Update renovates the existing kedaController object with kedaController definition in builder.
+func (builder *Builder) Update() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating configmap %s in namespace %s",
```
Comment (achuzhoy) [inline pkg/configmap/configmap.go:166]:
```suggestion
	glog.V(100).Infof("Updating kedaController %s in namespace %s",
```

**Comment by achuzhoy** [comment]
The title is "secret pkg small", but the change is to configmap

**Comment by elenagerman** [comment]
> The title is "secret pkg small", but the change is to configmap

@achuzhoy @trewest I see now, seems like a wrong commit was pushed under secret-fix mr name. fixed now

**Comment by elenagerman** [comment]
> > The title is "secret pkg small", but the change is to configmap
> 
> @achuzhoy @trewest I see now, seems like a wrong commit was pushed under secret-fix mr name. fixed now

I have 9 opened MRs + working on code for eco-gotest. the code viewer is crashing all the time. probably I pushed code from the wrong branch accidentally.

**Comment by elenagerman** [comment]
for configmap I have separate mr, all fixes were done there

**Review by trewest — APPROVED** [review]
/lgtm

**Comment by achuzhoy** [comment]
@elenagerman  you plan on adding the test for WithStringData in a subsequent PR?

**Comment by elenagerman** [comment]
> @elenagerman you plan on adding the test for WithStringData in a subsequent PR?

@achuzhoy I'm not sure about it, I can take care about it later, when secret unit test mr will be merged. @kononovn ask me do not implement secret unit test part since there is another person is working on it now. But I cannot stop my job and wait until unit test mr will be in.

moreover, I implemented secret unit test even twice already; I just forgot that Nikita ask me remove it for the first time and all implemented methods tests were there. So, actually I have it ready and tested, but not included to this mr

**Review by achuzhoy — APPROVED** [review]

**Pull request merged**